system_setup/server/controllers/locale.py

```diff
@@ -30,3 +32,65 @@ def start(self):
                           .format(self.model.selected_language))
 
         super().start()
+
+    async def _run_locale_support_cmds(self, lang: str):
+        """ Final commands to run when a valid language is POST'ed. """
+        cmds = [["locale-gen"],
+                ["update-locale", "LANG={}".format(lang)]]
+        env = os.environ.copy()
+
+        cp = await arun_command(["check-language-support", "-l", lang[0:5]],
+                                env=env)
+        if cp.returncode == 0:
+            packages = cp.stdout.strip('\n').split(' ')
+            cmds.append(["apt", "install", "-y"] + packages)
+
+        if self.app.opts.dry_run:
+            for cmd in cmds:
+                log.debug('Would run: ' + ' '.join(cmd))
+        else:
+            for cmd in cmds:
+                cp = await arun_command(cmd, env=env)
+                if cp.returncode:
+                    log.debug('Command \"%s\" failed with return code %d',
+                              cp.args, cp.returncode)
+
+    async def POST(self, data: str):
+        if data == self.autoinstall_default or data == os.environ.get("LANG"):
+            await super().POST(data)
+            return
+
+        fileContents: str
+        fname = "locale.gen"
+        env = os.environ.copy()
+        if self.app.opts.dry_run:
+            # For debug purposes.
+            fname = ".subiquity/" + fname
+            await arun_command(['cp', '/etc/locale.gen', '.subiquity/'],
+                               env=env)
+        else:
+            fname = "/etc/" + fname
+
+        pendingWrite = False
+        with open(fname, "r") as localeGen:
+            # locale.gen is not so big.
+            fileContents = localeGen.read()
+            lineFound = fileContents.find(data)
```

Comment:
Could we have "data" content being a substring like "pt" for ### pt_PT?
How do you handle as well 
```
# pt_BR ISO-8859-1
# pt_BR.UTF-8 UTF-8
```
As we probably want only the UTF-8 version anyway?

Reply:
Altered to regex.